tests/examples/shims/keygrabber.lua

```diff
@@ -1,11 +1,60 @@
 
-local keygrabber = {
-    run = function() end,
-    stop = function() end,
-    is_running = function() return false end,
+local keygrabber = nil
+
+local function run(grabber)
+    rawset(keygrabber, "_current_grabber", grabber)
+end
+
+local function stop()
+    keygrabber._current_grabber = nil
+end
+
+keygrabber = {
+    _run             = run,
+    stop             = stop,
+    isrunning        = function() return keygrabber._current_grabber ~= nil end,
+    _current_grabber = nil
```

Comment:
Why the `rawset` and why the mix between inline functions and separate functions? Couldn't `_current_grabber` simply be saved in a local?

Reply:
Nope, `root` need to have a pointer to send the fake_input to the grabber